rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -781,6 +789,64 @@ def preprocess_train_data(self, training_data: TrainingData) -> RasaModelData:
     def _check_enough_labels(model_data: RasaModelData) -> bool:
         return len(np.unique(model_data.get(LABEL_KEY, LABEL_SUB_KEY))) >= 2
 
+    def train_chunk(
+        self,
+        data_chunk_files: List[DataChunkFile],
+        config: Optional[RasaNLUModelConfig] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Trains this component using the list of data chunk files.
+
+        Args:
+            data_chunk_files: List of data chunk files.
+            config: The model configuration parameters.
+        """
+        if not data_chunk_files:
+            return
+
+        # TODO:
+        #  we need to make sure that we can actually train on this data, e.g. see
+        #  checks inside train().
+        #  Do we have any intent examples to train on?
+        #  Do we have any response examples to train on?
+        #  Do we have any entity examples to train on?
+        #  Do we have enough training examples, e.g. enough labels?
+
+        def _load_data_func(file_path: Path) -> RasaModelData:
+            training_data_chunk = TrainingDataChunk.load_chunk(file_path)
```

Comment:
Is this the right place to load the training data from files? I wonder if the data should be loaded before it's passed to the classifiers.

Reply:
Maybe some background information: We currently are not able to train on large datasets due to out of memory issues. That's why we are introducing a new command called `rasa train --number-of-chunks <n>`. This divides the training data into multiple smaller chunks and featurizes them separately. During training we need to train on all data, e.g. one epoch of training should see all training data. As the training data at this point is saved in multiple files on disk, as we cannot have everything in memory at the same time, we pass the data files to the training method. During training we load one data file at a time to make sure we are not running out of memory. This has the benefit that we can (hopefully) train on larger datasets, but has the downside that training will take longer as we need to load data multiple times from files.

So to answer your question, yes this is the right place as we cannot load all the data before training.